src/main/java/favourite/ui/FavouriteView.java

```diff
@@ -0,0 +1,62 @@
+package favourite.ui;
+
+import favourite.interface_adapters.FavouriteController;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+/**
+ * The FavouriteView class in the UI layer implements View and ActionListener interfaces.
+ * Whenever a user interacts with the favourite button, this class runs the Favourite use case.
+ */
+public class FavouriteView extends JPanel implements ActionListener, PropertyChangeListener {
+    // the controllwe that gets triggered when a user interacts with the FavouriteView
+    private final FavouriteController controller;
+    // the id of the post being favourited/unfavourited
+    private final int postid;
+
+    /**
+     * Initializes FavouriteView
+     *
+     * @param postid id of the post being favourited/unfavourited
+     */
+    public FavouriteView(int postid, FavouriteController controller){
```

Comment:
The int being passed as a param in constructor seems to be valid. This requires that the MainView knows about all the controllers (favourite, makepost, comment, etc), however